### ISSUE TITLE
[docs]: Update contribution docs

### ISSUE THREAD

**Issue by holmanb** [canonical/cloud-init#4646]
# Documentation request

The code [contribution page](https://cloudinit.readthedocs.io/en/latest/development/contribute_code.html) could use some attention.

1. [Feature flags](https://cloudinit.readthedocs.io/en/latest/development/contribute_code.html#module-cloudinit.features) deserves its own page - that is not a code contribution-specific thing to document.
2. [Conventional commits](https://github.com/canonical/cloud-init/pull/4593) should be documented.
3. [Other resources](https://cloudinit.readthedocs.io/en/latest/development/contribute_code.html#other-resources) is not relevant to code contributions, find a new home.

**Comment by blackboxsw** [comment]
Agreed on these restructuring points.  I do think add some content on conventional commits docs in the first PR page would be a good refefence as the link in the wall of text in our pull request template is easily overlooked/ignored.
